test/using_enum.cpp

```diff
@@ -0,0 +1,100 @@
+//
+// extented example from https://en.cppreference.com/w/cpp/language/enum
+//
+#include <magic_enum.hpp>
+
+#include <cstdint>
+#include <iostream>
+
+// enum that takes 16 bits
+enum smallenum : std::int16_t { a, b, c };
+
+// color may be red (value 0), yellow (value 1), green (value 20), or blue (value 21)
+enum color { red, yellow, green = 20, blue };
+
+// altitude may be altitude::high or altitude::low
+enum class altitude : char {
+  high = 'h',
+  low = 'l',  // trailing comma only allowed after CWG518
+};
+
+// the constant d is 0, the constant e is 1, the constant f is 3
+enum { d, e, f = e + 2 };
+
+// enumeration types (both scoped and unscoped) can have overloaded operators
+std::ostream& operator<<(std::ostream& os, color c) {
+#if 1
+  os << magic_enum::enum_name(c);
```

Comment:
magic_enum includes stream operators. using namespace magic_enum::ostream_operators; // out-of-the-box ostream operators for enums.